File: generator_php-symfony/package/src/Model/edxl/Content.php
<?php

namespace App\Model\edxl;

class Content
{
    /**
     * @var ContentObject
     *
     * // @JMS\Type("App\Model\edxl\ContentObject")
     * // @JMS\SerializedName("contentObject")
     * // @Assert\NotNull
     */
    private ?ContentObject $contentObject = null;

    public function __construct(?ContentObject $contentObject = null)
    {
        $this->contentObject = $contentObject;
    }

    public function getContentObject(): ?ContentObject
    {
        return $this->contentObject;
    }

    public function setContentObject(?ContentObject $contentObject): void
    {
        $this->contentObject = $contentObject;
    }

    public function __toString(): string
    {
        return sprintf(
            "Content{contentObject=%s}",
            $this->contentObject ? $this->contentObject->__toString() : 'null'
        );
    }
}

File: src/main/java/com/hubsante/model/edxl/Content.java
/**
 * Copyright © 2023-2026 Agence du Numerique en Sante (ANS)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.hubsante.model.edxl;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;

import java.util.Objects;

@JsonAutoDetect(
        fieldVisibility = JsonAutoDetect.Visibility.ANY,
        getterVisibility = JsonAutoDetect.Visibility.NONE
)
public class Content {

    @JacksonXmlProperty(localName = "xlink:type", isAttribute = true)
    public String getXmlns() {
        return "resource";
    }

    @JsonProperty(value = "contentObject", required = true)
    private ContentObject contentObject;

    public Content() {
    }

    public Content(ContentObject contentObject) {
        this.contentObject = contentObject;
    }

    public ContentObject getContentObject() {
        return contentObject;
    }

    public void setContentObject(ContentObject contentObject) {
        this.contentObject = contentObject;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Content content = (Content) o;
        return Objects.equals(contentObject, content.contentObject);
    }

    @Override
    public int hashCode() {
        return Objects.hash(contentObject);
    }

    @Override
    public String toString() {
        return "Content{" +
                "contentObject=" + contentObject +
                '}';
    }
}
